### PR TITLE
[6.13.z] Add tests for Facts component

### PR DESCRIPTION
**Pull request by Satellite-QE**
Cherrypick of PR: https://github.com/SatelliteQE/robottelo/pull/14933

### Problem Statement
Missing following test scenarios for Facts:

- Update facts for client (CLI and UI)

### Solution
Added test scenario for update client facts

### CODE REVIEW

**Comment by Satellite-QE** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k 'test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts'
airgun: 1437

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k 'test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts'
airgun: 1437

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7531
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts --external-logging
Test Result : ===== 1 failed, 1 passed, 8 deselected, 53 warnings in 1497.12s (0:24:57) ======
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k 'test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts'
airgun: 1445

File: tests/foreman/ui/test_fact.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    :customerscenario: true
+    """
+    with module_target_sat.ui_session() as session:
+        session.organization.select(module_entitlement_manifest_org.name)
+        session.location.select(module_location.name)
+        cmd = session.host.get_register_command(
+            {
+                'general.activation_keys': module_activation_key.name,
+                'general.insecure': True,
+            }
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_fact.py:50]: @shweta83 Looking at the PRT failure in UI, I remember we need to select operating_system here for 6.13.z which was mandatory field earlier in register host form
https://github.com/SatelliteQE/robottelo/blob/6.13.z/tests/foreman/ui/test_registration.py#L177

File: tests/foreman/cli/test_fact.py [hunk 2]
```diff
@@ -61,3 +63,55 @@ def test_negative_list_by_name(module_target_sat):
     assert (
         module_target_sat.cli.Fact().list(options={'search': f'fact={gen_string("alpha")}'}) == []
     )
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_fact.py:69]:
```suggestion
@pytest.mark.no_containers
@pytest.mark.pit_client
@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
```

**Comment by Gauravtalreja1** [comment]
@shweta83 Could you please update this PR?

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k 'test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts'
airgun: 1445

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8180
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts --external-logging
Test Result : ===== 1 failed, 1 passed, 8 deselected, 70 warnings in 1973.13s (0:32:53) ======
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k 'test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts'
airgun: 1445

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8184
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts --external-logging
Test Result : ===== 1 failed, 1 passed, 8 deselected, 73 warnings in 2344.61s (0:39:04) ======
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8187
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k test_positive_update_client_facts_verify_imported_values or test_positive_upload_host_facts --external-logging
Test Result : ===== 1 failed, 1 passed, 8 deselected, 78 warnings in 1739.75s (0:28:59) ======
```

**Comment by shweta83** [comment]
This test is trigger: test-robottelo
pytest: tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k 'test_positive_facts_end_to_end or test_positive_upload_host_facts'
airgun: 1445

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8198
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_fact.py tests/foreman/ui/test_fact.py -k test_positive_facts_end_to_end or test_positive_upload_host_facts --external-logging
Test Result : ========== 2 passed, 8 deselected, 72 warnings in 2396.14s (0:39:56) ===========
```

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**